gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationMapper.java

```diff
@@ -0,0 +1,95 @@
+package uk.nhs.adaptors.pss.translator.mapper.medication;
+
+import static uk.nhs.adaptors.pss.translator.util.ResourceUtil.generateMeta;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.IdType;
+import org.hl7.fhir.dstu3.model.Medication;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.v3.CD;
+import org.hl7.v3.RCMRMT030101UK04Consumable;
+import org.hl7.v3.RCMRMT030101UK04ManufacturedProduct;
+import org.hl7.v3.RCMRMT030101UK04Material;
+import org.hl7.v3.RCMRMT030101UK04MedicationStatement;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import uk.nhs.adaptors.pss.translator.mapper.CodeableConceptMapper;
+import uk.nhs.adaptors.pss.translator.service.IdGeneratorService;
+
+@Service
+@AllArgsConstructor
+public class MedicationMapper {
+    private static final String MEDICATION_URL
+        = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Medication";
+    private static final String HYPHEN = "-";
+    private static final Map<String, String> MEDICATION_IDS = new HashMap<>();
+
+    private CodeableConceptMapper codeableConceptMapper;
+    private IdGeneratorService idGeneratorService;
+
+    protected Medication createMedication(RCMRMT030101UK04Consumable consumable) {
+        if (hasManufacturedMaterial(consumable)) {
+            CD code = consumable.getManufacturedProduct().getManufacturedMaterial().getCode();
+
+            Medication medication = new Medication();
+            medication.setId(getMedicationId(code));
+            medication.setMeta(generateMeta(MEDICATION_URL));
+            medication.setCode(codeableConceptMapper.mapToCodeableConcept(code));
```

Comment:
can chain these (if you find this more readable then can leave it)

Reply:
Apologies, but could use the util you've made for this and add it to all the mappers that use it?

Reply:
Will leave as is, .setCode is not a method in Resource.class which is the return type once we start chaining methods

Reply:
have changed the other mappers to use meta generation however have left identifiers as they all seem to do it slightly differently.

Reply:
Think theres a ticket for the identifier source id stuff anyway so will likely be covered in that. Cheers!